solidity/contracts/isms/warp-route/AmountRoutingIsm.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+// ============ External Imports ============
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+
+// ============ Internal Imports ============
+import {AbstractRoutingIsm} from "../routing/AbstractRoutingIsm.sol";
+import {IInterchainSecurityModule} from "../../interfaces/IInterchainSecurityModule.sol";
+import {Message} from "../../libs/Message.sol";
+import {TypeCasts} from "../../libs/TypeCasts.sol";
+import {PackageVersioned} from "../../PackageVersioned.sol";
+import {TokenMessage} from "../../token/libs/TokenMessage.sol";
+
+/**
+ * @title AmountRoutingIsm
+ */
+contract AmountRoutingIsm is AbstractRoutingIsm, Ownable, PackageVersioned {
+    using Message for bytes;
+    using TokenMessage for bytes;
+    using Address for address;
+
+    IInterchainSecurityModule public immutable lower;
```

Comment:
Idk that we have a convention, but wonder if naming it lowerISM and upperISM makes it more descriptive?